core/src/main/java/org/infinispan/configuration/cache/DeadlockDetectionConfiguration.java

```diff
@@ -30,14 +33,20 @@ static AttributeSet attributeDefinitionSet() {
    /**
     * Time period that determines how often is lock acquisition attempted within maximum time
     * allowed to acquire a particular lock
+    *
+    * @deprecated Since 9.0, always returns {@code -1}.
     */
+   @Deprecated
    public long spinDuration() {
-      return attributes.attribute(SPIN_DURATION).get();
+      return spinDuration.get();
```

Comment:
@rvansa Agree, we should keep it. I'm not sure about adding another @Deprecated annotation, because the class already has it. I added it for the fluent methods, but only because they can be used without referencing the class.